src/plugins/run/annotations.ts

```diff
@@ -0,0 +1,203 @@
+          // finalize tabContainer
+          processed.push(tabPlaceholder);
+
+          // reset tabPlaceholder
+          inTab = false;
+          tabPlaceholder = {
+            type: 'tabs',
+            data: {
+              hName: 'tabs',
+            },
+            children: [
+              {
+                type: 'tab',
+                data: {
+                  hName: 'tab',
+                },
+                children: [],
+              },
+            ],
+          };
+
+          continue;
+        }
+      }
+
+      if (inTab) {
+        // if we're in a tab, push this node as a child of the last tab
+        const size = tabPlaceholder.children.length;
+        if (tabPlaceholder.children[size - 1]) {
+          (tabPlaceholder.children[size - 1].children as MDAST.Content[]).push(processNode(node, anno));
+        }
+      } else if (Object.keys(anno).length > 0 && next) {
+        // annotations apply to next node, process next node now and skip next iteration
+        processed.push(processNode(next, anno));
+        skipNext = true;
+      } else {
+        processed.push(processNode(node));
+      }
+    }
+
+    return {
+      ...root,
+      children: processed,
+    };
+  };
+};
+
+function captureAnnotations<T extends Dictionary<any>>(node: MDAST.Content | undefined): T | {} {
+  if (!node || !node.value) {
+    return {};
+  }
+
+  if (
+    // @ts-expect-error
+    node.type === 'mdxFlowExpression' &&
+    // @ts-expect-error
+    (node.value as string).startsWith('/*') &&
+    // @ts-expect-error
+    (node.value as string).endsWith('*/')
+  ) {
+    // remove comments and whitespace
+    // @ts-expect-error
+    const raw = (node.value as string)
+      // @ts-expect-error
+      .substr('/*'.length, (node.value as string).length - '*/'.length - '/*'.length)
+      .trim();
+
+    // load contents of annotation into yaml
+    try {
+      const contents = parse<T>(raw);
+      if (typeof contents === 'object') {
+        for (const key in contents) {
+          if (typeof contents[key] === 'string') {
+            // babel will crap out if certain characters, like ", are not escaped
+            const escapedContent = contents[key].replace('"', '%22');
+            contents[key] = escapedContent as any;
+          }
+        }
+
+        // annotations must be objects, otherwise it's just a regular ol html comment
+        return contents;
+      }
+    } catch (error) {
+      // ignore invalid YAML
+    }
+  } else if (
+    node.type === 'html' &&
+    (node.value as string).startsWith('<!--') &&
+    (node.value as string).endsWith('-->')
+  ) {
+    // remove comments and whitespace
+    const raw = (node.value as string)
+      .substr('<!--'.length, (node.value as string).length - '-->'.length - '<!--'.length)
+      .trim();
+
+    // load contents of annotation into yaml
+    try {
+      const contents = parse<T>(raw);
+      if (typeof contents === 'object') {
```

Comment:
what if it's a `null`? :P
```suggestion
      if (typeof contents === 'object' && contents !== null) {
```

Reply:
Ugh I always forget - good point

Reply:
Wait nvm - this is one of those "already in master" ones (https://github.com/stoplightio/markdown/blob/master/src/reader/transformers/to-spec.ts#L22). But doesn't matter, good point - will improve.